### PR TITLE
[CORE] Test opset1::Concat shape inference with op reuse

### PR DESCRIPTION
**Pull request by barnasm1**
### Details:
 - Add new tests to cover case with reuse of pre-created operator

### Tickets:

 - *[CVS-94507](https://jira.devtools.intel.com/browse/CVS-94507)*

### CODE REVIEW

**Comment by praasz** [comment]
@jane-intel could check if anything from ticket [CVS-94507](https://jira.devtools.intel.com/browse/CVS-94507) is missing. I think current shape inference works correctly and recalculate concatenation axis on each use. Added test to verify it.

**Review by praasz — APPROVED** [review]

File: src/plugins/intel_cpu/tests/unit/shape_inference_test/concat_shape_inference_test.cpp
[the remark this answers is not in the document]
Comment (praasz) [reply src/plugins/intel_cpu/tests/unit/shape_inference_test/concat_shape_inference_test.cpp:1]: Parametrized test always to only one inference on always created Concat operator. Additional test re-use created operator (with dynamic inputs) and check if positive value of concatenation is correctly calculated when inputs have different ranks (this axis will have different value on each use).

**Comment by jane-intel** [comment]
@praasz the problem was about this:
https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/concat.cpp#L45
https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/concat.hpp#L43
https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/concat.hpp#L60:L63

We probably need to remove the normalized axis from the node and return the newly normalized axis on each call of get_concatenation_axis

**Comment by praasz** [comment]
> @praasz the problem was about this: https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/concat.cpp#L45 https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/concat.hpp#L43 https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/concat.hpp#L60:L63
> ...

Sure we can review if setter/getter could be marked as deprecated, I assume deprecation message should be like removed in 2025.0?
The use of this concat axis could be reduced as `ov::Shape` and `ov::PartialShape` support negative index to get dimensions.

File: src/plugins/intel_cpu/tests/unit/shape_inference_test/concat_shape_inference_test.cpp
[the remark this answers is not in the document]
Comment (mlukasze) [reply src/plugins/intel_cpu/tests/unit/shape_inference_test/concat_shape_inference_test.cpp:1]: @maxnick is your question a blocker to merge it, or can we process?

**Pull request merged**